internal/provider/common.go

```diff
@@ -9,30 +10,48 @@ import (
 	"github.com/containerd/containerd/platforms"
 	"github.com/crazy-max/diun/v4/internal/model"
 	"github.com/crazy-max/diun/v4/pkg/registry"
-	"github.com/pkg/errors"
 )
 
 var (
 	metadataKeyChars  = `a-zA-Z0-9_`
 	metadataKeyRegexp = regexp.MustCompile(`^[` + metadataKeyChars + `]+$`)
+	ErrInvalidLabel   = errors.New("invalid label error")
 )
 
+func UpdateMap(m1, m2 map[string]string) map[string]string {
+	if m1 == nil && m2 != nil {
+		return m2
+	}
+
+	if m1 != nil && m2 == nil {
+		return m1
+	}
+
+	if m1 == nil && m2 == nil {
+		return nil
+	}
+
+	for key, value := range m2 {
+		m1[key] = value
+	}
+
+	return m1
+}
+
 // ValidateImage returns a standard image through Docker labels
-func ValidateImage(image string, metadata, labels map[string]string, watchByDef bool) (img model.Image, err error) {
+func ValidateImage(image string, metadata, labels map[string]string, watchByDef bool, imageDefaults model.Image) (img model.Image, err error) {
```

Comment:
I wonder if we could instead pass image as `model.Image` directly so we can avoid `watchByDef` and `imageDefaults`

Reply:
I think you still need `watchByDef` since there is no equivalent within `model.Image`.

That said, I do think a `ValidateImage` function would be better if it was merely validating and not constructing an Image. That would be a significant change as it would change the nature of this function. Could that be discussed and possibly implemented in a new PR?